catalogue/webapp/test/requesting.test.js

```diff
@@ -0,0 +1,28 @@
+describe('logged out', () => {
+  beforeAll(async () => {
+    await page.setCookie({
+      name: 'toggle_stacksRequestService',
+      value: 'true',
+      domain: 'localhost',
+      path: '/',
+    });
+    await page.goto('http://localhost:3000/works/h6tcdtrt');
+  });
+
+  it('contains a login link', async () => {
+    const cognitoUrl = 'https://id.wellcomecollection.org/oauth2';
+
+    const anchors = await page.$$('a');
+    const jsHandles = await Promise.all(
+      anchors.map(anchor => anchor.getProperty('href'))
+    );
+
+    const hrefs = await Promise.all(
+      jsHandles.map(handle => handle.jsonValue())
+    );
+
+    const loginUrls = hrefs.filter(href => href.match(cognitoUrl));
+
+    await expect(loginUrls.length).toEqual(1);
```

Comment:
This fails because the link is added to the page asynchronously. Haven't figured out how to address that.

Reply:
https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#pagewaitforselectororfunctionortimeout-options-args
❓

Reply:
There's waitForSelector and a whole load others for dealing with asynchronicity.